File: benchmark/output/lua-love/gemini/verlet_rope.lua
local node_xs = {}
local node_ys = {}
local prev_xs = {}
local prev_ys = {}
local pin_x = 200.0
local pin_y = 200.0
local speed = 200.0
local rest_len = 40.0
local num_nodes = 10

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Verlet Rope")
    for i = 0, num_nodes - 1 do
        local xv = 200.0 + i * 40.0
        table.insert(node_xs, xv)
        table.insert(prev_xs, xv)
        table.insert(node_ys, 200.0)
        table.insert(prev_ys, 200.0)
    end
end

function love.update(dt)
    if love.keyboard.isDown("left") then pin_x = pin_x - speed * dt end
    if love.keyboard.isDown("right") then pin_x = pin_x + speed * dt end
    if love.keyboard.isDown("up") then pin_y = pin_y - speed * dt end
    if love.keyboard.isDown("down") then pin_y = pin_y + speed * dt end

    node_xs[1] = pin_x
    node_ys[1] = pin_y
    prev_xs[1] = pin_x
    prev_ys[1] = pin_y

    for i = 2, num_nodes do
        local vx = node_xs[i] - prev_xs[i]
        local vy = node_ys[i] - prev_ys[i]
        prev_xs[i] = node_xs[i]
        prev_ys[i] = node_ys[i]
        node_xs[i] = node_xs[i] + vx * 0.99
        node_ys[i] = node_ys[i] + vy * 0.99 + 400.0 * dt * dt
    end

    for iter = 1, 3 do
        for i = 1, num_nodes - 1 do
            local dx = node_xs[i + 1] - node_xs[i]
            local dy = node_ys[i + 1] - node_ys[i]
            local dist = math.sqrt(dx * dx + dy * dy)
            if dist > 0.1 then
                local diff = (dist - rest_len) / dist * 0.5
                if i == 1 then
                    node_xs[i + 1] = node_xs[i + 1] - dx * diff * 2.0
                    node_ys[i + 1] = node_ys[i + 1] - dy * diff * 2.0
                else
                    node_xs[i] = node_xs[i] + dx * diff
                    node_ys[i] = node_ys[i] + dy * diff
                    node_xs[i + 1] = node_xs[i + 1] - dx * diff
                    node_ys[i + 1] = node_ys[i + 1] - dy * diff
                end
            end
        end
    end
end

function love.draw()
    for i = 1, num_nodes do
        love.graphics.circle("fill", node_xs[i], node_ys[i], 5.0)
    end
    for i = 1, num_nodes - 1 do
        for j = 0, 4 do
            local lt = j / 5.0
            local lx = node_xs[i] + (node_xs[i+1] - node_xs[i]) * lt
            local ly = node_ys[i] + (node_ys[i+1] - node_ys[i]) * lt
            love.graphics.circle("fill", lx, ly, 2.0)
        end
    end
    love.graphics.print("Arrows move anchor", 10.0, 10.0)
end

File: vibe-lang/benchmark/output/python-pygame/gemini/verlet_rope.py
import pygame
import sys
import math

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Verlet Rope Simulation")
clock = pygame.time.Clock()

# Game state variables
node_xs = []
node_ys = []
prev_xs = []
prev_ys = []
pin_x, pin_y = 200.0, 200.0
rest_len = 40.0
speed = 200.0

# Initialize nodes
for i in range(10):
    node_xs.append(200.0 + float(i) * 40.0)
    prev_xs.append(200.0 + float(i) * 40.0)
    node_ys.append(200.0)
    prev_ys.append(200.0)

font = pygame.font.Font(None, 36)

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    # Update logic
    keys = pygame.key.get_pressed()
    if keys[pygame.K_LEFT]: pin_x -= speed * dt
    if keys[pygame.K_RIGHT]: pin_x += speed * dt
    if keys[pygame.K_UP]: pin_y -= speed * dt
    if keys[pygame.K_DOWN]: pin_y += speed * dt

    node_xs[0] = pin_x
    node_ys[0] = pin_y
    prev_xs[0] = pin_x
    prev_ys[0] = pin_y

    # Verlet integration
    for i in range(1, 10):
        vx = node_xs[i] - prev_xs[i]
        vy = node_ys[i] - prev_ys[i]
        prev_xs[i] = node_xs[i]
        prev_ys[i] = node_ys[i]
        node_xs[i] = node_xs[i] + vx * 0.99
        node_ys[i] = node_ys[i] + vy * 0.99 + 400.0 * dt * dt

    # Constraint solve
    for _ in range(3):
        for i in range(9):
            dx = node_xs[i + 1] - node_xs[i]
            dy = node_ys[i + 1] - node_ys[i]
            dist = math.sqrt(dx * dx + dy * dy)
            if dist > 0.1:
                diff = (dist - rest_len) / dist * 0.5
                if i == 0:
                    node_xs[i + 1] -= dx * diff * 2.0
                    node_ys[i + 1] -= dy * diff * 2.0
                else:
                    node_xs[i] += dx * diff
                    node_ys[i] += dy * diff
                    node_xs[i + 1] -= dx * diff
                    node_ys[i + 1] -= dy * diff

    # Draw logic
    screen.fill((0, 0, 0))
    
    # Draw nodes
    for i in range(10):
        pygame.draw.circle(screen, (255, 255, 255), (int(node_xs[i]), int(node_ys[i])), 5)
    
    # Draw segments
    for i in range(9):
        for j in range(5):
            lt = float(j) / 5.0
            x = node_xs[i] + (node_xs[i+1] - node_xs[i]) * lt
            y = node_ys[i] + (node_ys[i+1] - node_ys[i]) * lt
            pygame.draw.circle(screen, (255, 255, 255), (int(x), int(y)), 2)

    text = font.render("Arrows move anchor", True, (255, 255, 255))
    screen.blit(text, (10, 10))

    pygame.display.flip()

pygame.quit()
sys.exit()